### DISCUSSION TITLE
Setting AWS credentials under Lambda environment

### DISCUSSION THREAD

**Discussion by nenadfspartans** [imgproxy/imgproxy#1307] [Q&A]
Hi guys, didn't  want to open issue because its probably on my end.
I have already deployed imgproxy as image service on lightsail for about a year and its working well. I am using infisical tool to pump environment variables into it. What I am trying to do now is convert that service to be deployed under lambda containe but I am constantly getting s3 error while AWS credentials are defined under infisical for this project, I could leave it as is without credentials because my bucket is public but I want to know why is it failing so can I set it up properly in case of private bucket.

This is log from lambda when aws keys are defined (access_key_id, access_key_secret):

[31mERROR  [0m [2024-07-08T06:52:45Z] Completed in 167.422465ms /pkpzKONn3S7Al5-YNbCl5EvYL_cKUdceDZ8rp6d-qC0/c:775:775:nowe:180:24/czM6Ly9oaWRkZW4tdGVzdC8xNzE5ODMzNTg0NjAwXzBfMWk5N3FoNDlpYXRfY3JvcC5qcGc?  [1mrequest_id[0m=68644a6c-77ac-446e-9d83-588853f6a329 [1mmethod[0m=GET [1mstatus[0m=404 [1mclient_ip[0m=xx.xxx.xxx.xxx [1merror[0m="Can't download source image: Status: 400; operation error S3: GetObject, https response error StatusCode: 400, RequestID: EZE939S7F43SEFDJ, HostID: oZSgV6sipnR7iGpkTwKs9NYVG9pDE6UY3x/OIN4QUUvOvuJcqQRnHWBoSaOEkq72i+j3q5+5/k0=, api error InvalidToken: The provided token is malformed or otherwise invalid."

I even tried to give full s3 permission to that user and it still fails at this point, If I run container outside of lambda it works well. My old service under lightsail also works well with defined aws credentials.

Thanks!

**Comment by nenadfspartans** [comment]
Very weird but setting AWS_SESSION_TOKEN to empty environment variable fixes issue. Maybe document this somewhere?

**Comment by DarthSim** [comment]
Hey @nenadfspartans!

There's no need to set AWS credentials when running imgproxy on AWS. Instead, you should provide access to AWS services through AWS policies.
